vehicles/src/modules/permit-application-payment/payment/payment.service.ts

```diff
@@ -710,12 +710,12 @@ export class PaymentService {
     );
 
     if (application.permitStatus === ApplicationStatus.VOIDED) {
-      const oldAmount = calculatePermitAmount(permitPaymentHistory);
-      if (oldAmount > 0) return -oldAmount;
-      return oldAmount;
+      const newAmount = permitFee(application);
+      return newAmount;
     }
     const oldAmount = calculatePermitAmount(permitPaymentHistory);
-    return permitFee(application, oldAmount);
```

Comment:
For ```calculatePermitAmount(permitPaymentHistory)```, is ```permitPaymentHistory``` filtered to only include successful transactions? Otherwise we would still run into the problem of amount mismatch due to duplicate transactions when calculating total old amount (for amend).

Reply:
okay, was thinking of creating another ticket for that, but will do it in this one.